modules/text2vec-databricks/module.go

```diff
@@ -0,0 +1,154 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package moddatabricks
+
+import (
+	"context"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/weaviate/weaviate/modules/text2vec-databricks/ent"
+
+	"github.com/weaviate/weaviate/usecases/modulecomponents/text2vecbase"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/weaviate/weaviate/entities/models"
+	"github.com/weaviate/weaviate/entities/modulecapabilities"
+	"github.com/weaviate/weaviate/entities/moduletools"
+	"github.com/weaviate/weaviate/modules/text2vec-databricks/clients"
+	"github.com/weaviate/weaviate/modules/text2vec-openai/vectorizer"
+	"github.com/weaviate/weaviate/usecases/modulecomponents/additional"
+)
+
+const (
+	Name = "text2vec-databricks"
+)
+
+func New() *DatabricksModule {
+	return &DatabricksModule{}
+}
+
+type DatabricksModule struct {
+	vectorizer                   text2vecbase.TextVectorizerBatch
+	metaProvider                 text2vecbase.MetaProvider
+	graphqlProvider              modulecapabilities.GraphQLArguments
+	searcher                     modulecapabilities.Searcher
+	nearTextTransformer          modulecapabilities.TextTransform
+	logger                       logrus.FieldLogger
+	additionalPropertiesProvider modulecapabilities.AdditionalProperties
+}
+
+func (m *DatabricksModule) Name() string {
+	return Name
+}
+
+func (m *DatabricksModule) Type() modulecapabilities.ModuleType {
+	return modulecapabilities.Text2MultiVec
+}
+
+func (m *DatabricksModule) Init(ctx context.Context,
+	params moduletools.ModuleInitParams,
+) error {
+	m.logger = params.GetLogger()
+
+	if err := m.initVectorizer(ctx, params.GetConfig().ModuleHttpClientTimeout, m.logger); err != nil {
+		return errors.Wrap(err, "init vectorizer")
+	}
+
+	if err := m.initAdditionalPropertiesProvider(); err != nil {
+		return errors.Wrap(err, "init additional properties provider")
+	}
+
+	return nil
+}
+
+func (m *DatabricksModule) InitExtension(modules []modulecapabilities.Module) error {
+	for _, module := range modules {
+		if module.Name() == m.Name() {
+			continue
+		}
+		if arg, ok := module.(modulecapabilities.TextTransformers); ok {
+			if arg != nil && arg.TextTransformers() != nil {
+				m.nearTextTransformer = arg.TextTransformers()["nearText"]
+			}
+		}
+	}
+
+	if err := m.initNearText(); err != nil {
+		return errors.Wrap(err, "init graphql provider")
+	}
+	return nil
+}
+
+func (m *DatabricksModule) initVectorizer(ctx context.Context, timeout time.Duration,
+	logger logrus.FieldLogger,
+) error {
+	databricksToken := os.Getenv("DATABRICKS_TOKEN")
```

Comment:
let's change the `DATABRICKS_TOKEN` to `DATABRICKS_APIKEY` this will be then inline with our naming convention in all other modules

Reply:
But in the databricks docs, it's called "Databricks Token". I figured we should follow their naming convention so that databricks users have a consistent experience.

What do you think?

Reply:
@antas-marcin ☝️

Reply:
Can you point me the docs?

Reply:
here, under requirements:

https://docs.databricks.com/en/machine-learning/foundation-models/index.html#requirements

Reply:
ok, let's leave it then